hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -624,6 +635,11 @@ public void run() {
             }
             partitionServiceLock.lock();
             try {
+                triggerRepartitioningWhenClusterStateAllowsMigration = !isMigrationAllowedByClusterState();
+                if (triggerRepartitioningWhenClusterStateAllowsMigration && logger.isFineEnabled()) {
+                    logger.fine("Migrations are not allowed yet, "
+                            + "repartitioning will be triggered when cluster state allows migrations.");
```

Comment:
We proceed to repartition here, even when `triggerRepartitioningWhenClusterStateAllowsMigration` is `true`. Is this intentional?

Reply:
Got it, thanks!